pkg/services/hcloud/server/server_test.go

```diff
@@ -339,3 +341,111 @@ var _ = Describe("Test ValidateLabels", func() {
 		}),
 	)
 })
+
+var _ = Describe("Test handleRateLimit", func() {
+	type testCaseHandleRateLimit struct {
+		hm           *infrav1.HCloudMachine
+		err          error
+		functionName string
+		errMsg       string
+		expectError  error
+		conditionSet bool
+	}
+
+	DescribeTable("Test handleRateLimit",
+		func(tc testCaseHandleRateLimit) {
+			err := handleRateLimit(tc.hm, tc.err, tc.functionName, tc.errMsg)
+			if tc.expectError != nil && hcloud.IsError(err, hcloud.ErrorCodeRateLimitExceeded) {
```

Comment:
I'm not sure whether you need this at all... isn't this replaced by what we have below?